### ISSUE TITLE
Smoother tree component animation

### ISSUE THREAD

**Issue by janett-baresel** [Esri/calcite-design-system#6863]
### Description

The tree component animates the expand and collapse action. I would like to request to animate the change of height of the container additionally to the expansion of the items.

Compare the calcite tree component animation:

https://user-images.githubusercontent.com/11048042/234509492-5af04143-7496-4b95-bfcc-04c823292857.mp4

With SV layer list animation:

https://user-images.githubusercontent.com/11048042/234529184-07247958-405e-4568-b0f8-0e4a191dcd69.mp4

The calcite tree animation doesn't smoothly animate the height change of the group. This results in a more clunky / less smooth animation.

### Acceptance Criteria

animate height of the container at same speed and timing as expansion/appearing of the tree child elements.

### Which Component

Tree

### Priority impact

p4 - not time sensitive

### Esri team

ArcGIS Scene Viewer

**monday.com sync:** #9363616200

**Comment by geospatialem** [comment]
Adding `design` for an initial assessment of the `tree` animation for consideration prior to development work/next steps.

**Comment by macandcheese** [comment]
Could we close this in favor of the more recent and more active related issues: https://github.com/Esri/calcite-design-system/issues/6473 https://github.com/Esri/calcite-design-system/issues/12669

cc @Esri/calcite-pes @Esri/calcite-designers

**Comment by geospatialem** [comment]
> Could we close this in favor of the more recent and more active related issues: [#6473](https://github.com/Esri/calcite-design-system/issues/6473) [#12669](https://github.com/Esri/calcite-design-system/issues/12669)
> 
> cc [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes) [@Esri/calcite-designers](https://github.com/orgs/Esri/teams/calcite-designers)

Agreed - closing based on [Adam's comment above](https://github.com/Esri/calcite-design-system/issues/6863#issuecomment-3746895985), in favor of https://github.com/Esri/calcite-design-system/issues/6473 and https://github.com/Esri/calcite-design-system/issues/12669
